Assets/Scripts/Player.cs

```diff
@@ -0,0 +1,63 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class Player : MonoBehaviour
+{
+    private const float ANIM_LOCK_DURATION = 0.05f;
+
+    [System.Serializable] public class PositionChangedEvent : UnityEvent<Vector2> {}
+
+    [SerializeField] private float moveSpeed = 1f;
+
+    private bool isAnimLocked;
+    private Vector3 prevFramePosition;
+
+    private YieldInstruction animLockInstruction;
+    private Animator animator;
+    private new Rigidbody2D rigidbody2D;
+
+    private PositionChangedEvent onPositionChanged = new PositionChangedEvent();
+
+    void Awake()
+    {
+        animLockInstruction = new WaitForSeconds(ANIM_LOCK_DURATION);
+        animator = GetComponent<Animator>();
+        rigidbody2D = GetComponent<Rigidbody2D>();
+    }
+
+    void Update()
+    {
+        float inputH = Input.GetAxisRaw("Horizontal");
+        float inputV = Input.GetAxisRaw("Vertical");
+        Vector3 inputDir = new Vector3(inputH, inputV).normalized;
```

Comment:
Is there a reason to use Vector3 as opposed to Vector2 here?

There's no functional difference as far as I can tell, but it would be slightly more clear to the reader.

Reply:
Nice catch, that was a remnant of a previous implementation. Updated.